src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,315 @@
+        }
+        backslash = c == '\\';
+      }
+    }
+    return pgTerm.toString();
+  }
+
+  /**
+   * CQL full text term to Postgres term.
+   * @see <a href="https://www.postgresql.org/docs/13/sql-syntax-lexical.html">
+   *   String Constants section</a>
+   *
+   * <p>At this stage masking is unsupported and rejected.</p>
+   * @param termNode which includes term and relation.
+   * @return Postgres term.
+   */
+  static String cqlTermToPgTermFullText(CQLTermNode termNode) {
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      // handle the CQL specials *, ?, ^, \\, rest are passed through as is
+      if (c == '*') {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op * unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '?') {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op ? unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '^') {
+        if (!backslash) {
+          throw new IllegalArgumentException("Anchor op ^ unsupported for: " + termNode.toCQL());
+        }
+      } else if (c != '\\' && backslash) {
+        pgTerm.append('\\');
+      }
+      if (c == '\\' && !backslash) {
+        backslash = true;
+      } else {
+        pgTerm.append(c);
+        if (c == '\'') {
+          pgTerm.append(c);
+        }
+        backslash = false;
+      }
+    }
+    return pgTerm.toString();
+  }
+
+  String handleTypeText(PgCqlField field, CQLTermNode termNode, boolean fullText) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String base = termNode.getRelation().getBase();
+    if (fullText) {
+      fullText = "=".equals(base) || "all".equals(base);
+    }
+    if (fullText) {
+      String pgTerm = cqlTermToPgTermFullText(termNode);
+      return "to_tsvector('" + language + "', " + field.getColumn() + ") @@ plainto_tsquery('"
+          + language + "', '" + pgTerm + "')";
+    }
+    return field.getColumn() + " " + basicOp(termNode)
+        + " '" +  cqlTermToPgTermExact(termNode) + "'";
+  }
+
+  String handleTypeNumber(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String cqlTerm = termNode.getTerm();
+    if (cqlTerm.isEmpty()) {
+      throw new IllegalArgumentException("Bad numeric for: " + termNode.toCQL());
+    }
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      switch (c) {
+        case '.':
+        case ',':
+        case '+':
+        case '-':
+          break;
+        default:
+          if (!Character.isDigit(c)) {
+            throw new IllegalArgumentException("Bad numeric for: " + termNode.toCQL());
+          }
+      }
+    }
```

Comment:
This code accepts strings that PostgreSQL doesn't accept and result in SQL parse errors: `1.2.3`, `0+++0` the [Devanagari numeral](https://en.wikipedia.org/wiki/Devanagari_numerals) `१`.
This code doesn't accept valid PostgreSQL numbers, for example `1e2`, `2E+9`.

We already have a well tested PostgreSQL number check:
https://github.com/folio-org/raml-module-builder/blob/v33.0.2/cql2pgjson/src/main/java/org/folio/cql2pgjson/util/Cql2SqlUtil.java#L10-L23
https://github.com/folio-org/raml-module-builder/blob/v33.0.2/cql2pgjson/src/test/java/org/folio/cql2pgjson/util/Cql2SqlUtilTest.java#L481-L520